packages/common/src/core/_internal/computeBaseProps.ts

```diff
@@ -0,0 +1,17 @@
+import { IItem } from "@esri/arcgis-rest-types";
+import { IHubItemEntity } from "../types";
+import { deriveLocationFromItem } from "../../content/_internal/internalContentUtils";
+
+/**
+ * Base property mapping for item backed entity types
+ * @param item IItem
+ * @param entity IHubItemEntity
+ * @returns
+ */
+export function computeBaseProps<EntityType extends Partial<IHubItemEntity>>(
+  item: IItem,
+  entity: EntityType
```

Comment:
This should probably be `IHubItemEntity` instead

Reply:
If we use `IHubItemEntity`, we will need to cast as specific types `as IHubProject`, etc.  The above approach uses generics, so the return type will always match up.

Reply:
Ahhhhhh ok, I was confused because we have an `EntityType` type already in the project (`IHubCatalog.ts`)